packages/discovery-provider/plugins/pedalboard/apps/solana-relay/src/routes/listen/listen.ts

```diff
@@ -148,6 +154,25 @@ export const validateListenSignature = async (
   return false
 }
 
+// rate limiter for after request validation
+export const listenRouteRateLimiter = async (params: { ip: string, trackId: string, logger?: Logger }): Promise<{ allowed: boolean }> => {
+  const { ip, trackId, logger } = params
+  const ipTrackConcatKey = `${ip}:${trackId}`
+
+  // consume and check rate limits
+  const ipLimit = await listensIpRateLimiter.checkLimit(ip)
+  const ipTrackLimit = await listensIpTrackRateLimiter.checkLimit(ipTrackConcatKey)
+
+  // merge limits and check if both allow passage
+  const allowed = ipLimit.allowed && ipTrackLimit.allowed
+
+  if (!allowed) {
+    logger?.info({ ipLimit, ipTrackLimit, ip }, "rate limit hit")
```

Comment:
Does this logger decorate messages with the endpoint by default? Just want to make sure that if we're searching for this in logs, it's clear that it's for this specific endpoint.
Might also want to know which track it's for.

Reply:
it does, it's a child logger passed in from the handler